crates/iota-data-ingestion-core/src/reader/fetch.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::BTreeMap,
+    ffi::OsString,
+    fmt::Display,
+    fs,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use iota_rest_api::{CheckpointData, Client};
+use iota_storage::blob::Blob;
+use iota_types::messages_checkpoint::CheckpointSequenceNumber;
+use notify::{RecursiveMode, Watcher};
+use object_store::{ObjectStore, path::Path as ObjectStorePath};
+use tokio::sync::mpsc;
+use tracing::{debug, info};
+
+use crate::{IngestionError, IngestionResult, MAX_CHECKPOINTS_IN_PROGRESS};
+
+pub type CheckpointResult = IngestionResult<(Arc<CheckpointData>, usize)>;
+
+/// Managing and processing checkpoint files in a directory.
+pub(crate) trait LocalCheckpointFileOps {
```

Comment:
`Ops` is ambiguous. What about renaming `LocalRead`?

Reply:
fixed: https://github.com/iotaledger/iota/pull/7424/commits/cbcfae532205b69d597fa4779babdd1d4ac0dc99